test/python/test_api.py

```diff
@@ -201,43 +201,47 @@ def _run_testing(x_0, s_0, b_0, rm_0, rv_0, m_0=0.1):
 
     def test_softmax_api(self):
```

Comment:
in the future, we can improve it as follows
```python
def softmax_api_test(dev)

def test_softmax_api_cpu(self)
   cpu = ...
   softmax_api_test(cpu)

@unittestskip(...)
def test_softmax_api_gpu(self)
   gpu = ...
   softmax_api_test(gpu)
```
In this way, the unittest will report if the error is from cpu test or gpu test. Otherwise, it will only report that the softmax_api has error.

Reply:
fixed